File: 01/rb/01.rb
#!/usr/bin/env ruby

a,b = File.readlines('input').map{|line| line.split.map &:to_i }.transpose

# part 1
p a.sort.zip(b.sort).sum{|x,y| (x-y).abs }

# part 2
p a.sum{|x| b.count(x) * x }

File: 01/hs/01.hs
#!/usr/bin/env runhaskell

import Data.List

part1 = sum . map diff . uncurry zip . both sort
    where both f (a,b) = (f a,f b)
          diff = abs . uncurry (-)

part2 (a,b) = sum $ a >>= (\x -> filter (==x) b)

main = do
    [a,b] <- transpose <$> map (map read . words) <$> lines <$> readFile "input"
    putStrLn . show $ part1 (a,b)
    putStrLn . show $ part2 (a,b)
